aleph/manage.py

```diff
@@ -504,6 +504,19 @@ def upgrade():
     # upgrade_collections()
 
 
+@cli.command()
+@click.option(
+    "--revision",
+    "-r",
+    default="-1",
+    show_default=True,
+    help="target revision number. Can also be 'head' or '-1'",
+)
+def downgrade_database(revision):
+    """Downgrade the database."""
+    downgrade_db(revision)
```

Comment:
Running database downgrades can be quite scary as they can delete data. The core alembic CLI has a dry-run option and I just checked and Flask-Migrate exposes that, too.

```suggestion
@cli.command()
@click.option(
    "--revision",
    "-r",
    default="-1",
    show_default=True,
    help="target revision number. Can also be 'head' or '-1'",
)
@click.option(
    "--sql",
    is_flag=False,
    is_flag=True,
    help="...",
)
def downgrade_database(revision, sql):
    """Downgrade the database."""
    downgrade_db(revision, sql)
```

Something like this should print the SQL statements the migration would execute without actually executing it, as well as the names of the migrations that will be rolled back.

This would allow us to double check that the downgrade is doing what we expect before actually executing it.

Reply:
When I tested this locally, I accidentally rolled back a different migration than I expected… 😄

Reply:
I made these changes:

- the default is a dry run, which prints a warning that this doesn't actually execute, the migration and the SQL statements and quits.
- you now have to pass `--execute` to actually run the migrations.